awx/main/dispatch/__init__.py

```diff
@@ -19,7 +19,17 @@ def get_task_queuename():
     if os.getenv('AWX_COMPONENT') == 'web':
         from awx.main.models.ha import Instance
 
-        return Instance.objects.filter(node_type__in=['control', 'hybrid']).order_by('?').first().hostname
+        return (
+            Instance.objects.filter(
+                node_type__in=['control', 'hybrid'],
+                node_state__in=['ready'],
+                enabled__in=['true'],
+            )
+            .order_by('?')
+            .first()
+            .hostname
```

Comment:
if you're only going to use the hostname property, you might as well add `.only('hostname')` for performance.

Reply:
added